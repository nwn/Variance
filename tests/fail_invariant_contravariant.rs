mod common;

use common::*;

fn invariant_fail_contravariant<'a>() {
    let _in: In<Lifetime<'static>> = In(
        Invariant::<Lifetime<'a>>::default(),
    );
}

fn main() {}
