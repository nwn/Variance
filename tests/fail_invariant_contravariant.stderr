error[E0308]: mismatched types
 --> $DIR/fail_invariant_contravariant.rs:7:9
  |
7 |         Invariant::<Lifetime<'a>>::default(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
  |
  = note: expected struct `type_variance::Invariant<common::Lifetime<'static>>`
             found struct `type_variance::Invariant<common::Lifetime<'a>>`
note: the lifetime `'a` as defined on the function body at 5:33...
 --> $DIR/fail_invariant_contravariant.rs:5:33
  |
5 | fn invariant_fail_contravariant<'a>() {
  |                                 ^^
  = note: ...does not necessarily outlive the static lifetime
